realm/realm-library/src/main/java/io/realm/internal/SharedGroupManager.java

```diff
@@ -169,10 +169,12 @@ public static boolean compact(RealmConfiguration configuration) {
         try {
             sharedGroup = new SharedGroup(
                     configuration.getPath(),
-                    SharedGroup.EXPLICIT_TRANSACTION,
+                    SharedGroup.IMPLICIT_TRANSACTION,
```

Comment:
We get a `LogicError` exception when you try to open the Realm with explicit transaction if it is already opened with implicit transaction. I haven't talked with the core team so I don't know if they are aware of it.